File: frontend/lib/models/rendez_vous.dart
class RendezVous {
  final int id;
  final int fideleId;
  final String type;
  final String sujet;
  final DateTime? dateSouhaitee;
  final String? heureSouhaitee;
  final String statut;
  final String? noteFidele;
  final String? notePasteur;
  final int? assigneAId;
  final DateTime? dateEffectif;
  final Map<String, dynamic>? fidele;
  final Map<String, dynamic>? assigneA;

  RendezVous({
    required this.id,
    required this.fideleId,
    this.type = 'pastoral',
    required this.sujet,
    this.dateSouhaitee,
    this.heureSouhaitee,
    this.statut = 'en_attente',
    this.noteFidele,
    this.notePasteur,
    this.assigneAId,
    this.dateEffectif,
    this.fidele,
    this.assigneA,
  });

  factory RendezVous.fromJson(Map<String, dynamic> json) {
    return RendezVous(
      id: json['id'],
      fideleId: json['fidele_id'],
      type: json['type'] ?? 'pastoral',
      sujet: json['sujet'],
      dateSouhaitee: json['date_souhaitee'] != null ? DateTime.parse(json['date_souhaitee']) : null,
      heureSouhaitee: json['heure_souhaitee']?.toString(),
      statut: json['statut'] ?? 'en_attente',
      noteFidele: json['note_fidele'],
      notePasteur: json['note_pasteur'],
      assigneAId: json['assigne_a'],
      dateEffectif: json['date_effectif'] != null ? DateTime.tryParse(json['date_effectif'].toString()) : null,
      fidele: json['fidele'],
      assigneA: json['assigne_a'] is Map ? json['assigne_a'] : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'fidele_id': fideleId,
        'type': type,
        'sujet': sujet,
        'date_souhaitee': dateSouhaitee?.toIso8601String().split('T')[0],
        'heure_souhaitee': heureSouhaitee,
        'note_fidele': noteFidele,
      };

  String get typeLabel => type == 'pastoral' ? 'Pastoral' : type == 'priere' ? 'Prière' : 'Autre';
  String get statutLabel {
    switch (statut) {
      case 'confirme': return 'Confirmé';
      case 'annule': return 'Annulé';
      case 'effectue': return 'Effectué';
      default: return 'En attente';
    }
  }
}

File: backend/temp_backend/app/Models/RendezVous.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class RendezVous extends Model
{
    use HasFactory;

    protected $table = 'rendez_vous';

    protected $fillable = [
        'fidele_id',
        'type',
        'sujet',
        'date_souhaitee',
        'heure_souhaitee',
        'statut',
        'note_fidele',
        'note_pasteur',
        'assigne_a',
        'date_effectif',
    ];

    protected $casts = [
        'date_souhaitee' => 'date',
        'date_effectif' => 'datetime',
    ];

    public function fidele()
    {
        return $this->belongsTo(Fidele::class);
    }

    public function assigneA()
    {
        return $this->belongsTo(User::class, 'assigne_a');
    }
}
